### PR TITLE
SSH key upload via UI

### PR DESCRIPTION
**Pull request by mahakporwal02**
## Rationale

[//]: # (Briefly explain the reason behind this change.)
Easier to upload SSH key via UI

#363 
## Changes
Changed permissions for adding ssh keys and create  UI in edit tab in updateUser

### CODE REVIEW

**Comment by mahakporwal02** [comment]
Screenshot from 2021-01-08 18-59-20
Screenshot from 2021-01-08 18-59-32
@rgaudin When I am applying black formatting to init.py the spaces in `permission=""` disappears
So should I add them anyways?

**Comment by rgaudin** [comment]
@kelson42 are you happy with the depicted workflow? Or would you prefer it to upload the RSA public key file directly?

**Comment by kelson42** [comment]
@rgaudin @mahakporwal02 I think this is good but:
* it should be a textarea (big enough so the user can somehow look to what he has copy/paste)
* A filter should be done to remove the carriage returns that a cut and paste might have introduced
* The result should be checked if it looks good, otherwise if will just fail later and the user will ask why

**Comment by mahakporwal02** [comment]
Screenshot from 2021-01-11 19-28-00

**Comment by mahakporwal02** [comment]
Screenshot from 2021-01-11 20-05-01

**Comment by kelson42** [comment]
@mahakporwal02 Do you expect new review from us now?

**Comment by mahakporwal02** [comment]
@kelson42 @rgaudin Yes, I expect a review

**Comment by kelson42** [comment]
@mahakporwal02 From the UI point-of-view it looks good. Thx!

File: dispatcher/frontend-ui/src/components/UpdateUser.vue [hunk 1]
```diff
@@ -38,6 +47,9 @@
     data() {
       return {
         form: {},
+        keyName: null, 
+        key: null,
```
Comment (rgaudin) [inline dispatcher/frontend-ui/src/components/UpdateUser.vue:51]: None of those declared vars are used anyway…

File: dispatcher/frontend-ui/src/components/UpdateUser.vue [hunk 2]
```diff
@@ -93,6 +105,26 @@
             parent.toggleLoader(false);
           });
       },
+      addKey() { // request token on API using credentials
+        let parent = this;
+
+        parent.working = true;
```
Comment (rgaudin) [inline dispatcher/frontend-ui/src/components/UpdateUser.vue:111]: Why not using similar code as for updating user? Using a parent.error seems unnecessary

File: dispatcher/frontend-ui/src/components/UpdateUser.vue [hunk 3]
```diff
@@ -93,6 +105,26 @@
...
+        parent.error = null;
+
+        const key = parent.form.key.replace(/[\n\r]+/g, '').trim();
+        const payload = {name: parent.form.keyName, key: key};
```
Comment (rgaudin) [inline dispatcher/frontend-ui/src/components/UpdateUser.vue:115]: You're not testing what you send here ; you can send empty values

File: dispatcher/frontend-ui/src/components/UpdateUser.vue [hunk 4]
```diff
@@ -93,6 +105,26 @@
...
+        parent.queryAPI('post', '/users/' + parent.user.username + '/keys', payload)
+            .then(function () {
+              parent.alertSuccess("Added!", "SSH key has been added.");
+              parent.$router.back();  // redirect
```
Comment (rgaudin) [inline dispatcher/frontend-ui/src/components/UpdateUser.vue:119]: Why going back? Just redirect to the user's profile or users list

File: dispatcher/frontend-ui/src/views/UserView.vue [hunk 5]
```diff
@@ -55,7 +55,7 @@
           </tbody>
         </table>
 
-        <table v-if="user.ssh_keys" class="table table-sm table-striped table-in-tab">
+        <table v-if="user.ssh_keys.length" class="table table-sm table-striped table-in-tab">
```
Comment (rgaudin) [inline dispatcher/frontend-ui/src/views/UserView.vue]: broke the display of user profile for users without SSH keys (most of them)

File: dispatcher/frontend-ui/src/components/UpdateUser.vue [hunk 6]
```diff
@@ -20,6 +20,15 @@
       <b-input class="mr-2" id="cu_password" placeholder="Password" required v-model="form.password"></b-input>
       <b-button type="submit" class="form-control" :disabled="!form.password" variant="primary">Change Password</b-button>
     </b-form>
+
+    <hr />
+
+    <b-form v-if=" form.role == 'worker'" inline @submit.prevent="addKey">
+      <label class="mr-2" for="cu_password">Add SSH Key</label>
```
Comment (rgaudin) [inline dispatcher/frontend-ui/src/components/UpdateUser.vue:27]: label assigned to wrong element

**Review by rgaudin — CHANGES_REQUESTED** [review]
Thanks for this. There are a number of small issues, mostly formatting and preciseness.
I am pushing a fix for those but please take a moment to understand those minor mistakes.

**Review by rgaudin — APPROVED** [review]

**Pull request merged**